src/toil/test/src/importExportFileTest.py

```diff
@@ -14,64 +14,85 @@
 
 from __future__ import absolute_import
 
-import uuid
-
+import random
 import shutil
-
-import os
 from toil.common import Toil
 from toil.job import Job
+from toil.leader import FailedJobsException
 from toil.test import ToilTest
 import tempfile
 
 
 class ImportExportFileTest(ToilTest):
     def setUp(self):
-        self.srcFile = "%s/%s" % (tempfile.mkdtemp(), uuid.uuid4())
+        self._tempBaseDir = tempfile.mkdtemp()
+        self.srcFile = "%s/%s" % (self._tempBaseDir, 'in')
         with open(self.srcFile, 'w') as f:
-            f.write(os.urandom(2**10))
+            f.write('Hello')
 
     def tearDown(self):
-        shutil.rmtree(self.srcFile.rsplit('/', 1)[0])
+        shutil.rmtree(self._tempBaseDir)
 
     def testImportExportFile(self):
         options = Job.Runner.getDefaultOptions(self._getTestJobStorePath())
         options.logLevel = "INFO"
-        job = HelloWorld()
+
+        with Toil(options) as toil:
+            jobStoreFileID = toil.importFile('file://' + self.srcFile)
+
+            jobStoreFileID = toil.run(HelloWorld(jobStoreFileID))
+            self._exportJobStoreFileID(toil, jobStoreFileID)
+
+    def testImportExportFileRestart(self):
+        options = Job.Runner.getDefaultOptions(self._getTestJobStorePath())
+        options.logLevel = "INFO"
+        options.clean = 'never'  # 'onSuccess'
+
         with Toil(options) as toil:
-            jobStoreFileID = toil.importFile('file://%s' % self.srcFile)
-            assert toil.jobStore.fileExists(jobStoreFileID)
+            jobStoreFileID = toil.importFile('file://' + self.srcFile)
 
-            destPath = '%s/%s' % (self.srcFile.rsplit('/', 1)[0], uuid.uuid4())
-            toil.exportFile(jobStoreFileID, 'file://%s' % destPath)
-            assert os.path.exists(destPath)
+            while True:
+                try:
+                    jobStoreFileID = toil.run(HelloWorldFails(jobStoreFileID))
+                    break
```

Comment:
Repeat the entire test 10 times, i.e. add another outer loop, count the executions of the inner loop's body (without ever resetting the counter) and assert that the counter is > 10 at the end. To speed things up you could also break out of the outer loop as soon as you observe the inner loop body to be executed more than once per iteration of the outer loop.